web/client/components/misc/enhancers/infiniteScroll/loadMore.js

```diff
@@ -16,8 +16,8 @@ const Rx = require('rxjs');
  * @param  {function} loadPage           A function that returns the observable that emits the page loaded. the event emitted must have at least { items [ ...items of the new page], total: the total number of results}
  * @return {Observable}                    Stream of props {with items, loading, error, total}
  */
-const loadMoreStream = (initialStream$, loadMore$, loadPage, {dataProp = "items", throttleTime = 500} = {}) =>
-    initialStream$.switchMap(searchParams =>
+const loadMoreStream = (initialStream$, loadMore$, loadPage, {dataProp = "items", initialStreamDebounce = 0, throttleTime = 500} = {}) =>
+    initialStream$.take(1).concat(initialStream$.debounceTime(initialStreamDebounce)).switchMap(searchParams =>
```

Comment:
can you explain the reason this take + concat change? Maybe a comment can help understanding it?

Reply:
It makes the first initialStream event to be executed instantly regardless of the initialStreamDebounce value. This is useful because the first event is dispatched when a dialog, a panel, etc. with infinite scroll is opened, debounce is not needed there.